stor/obs.py

```diff
@@ -142,12 +142,35 @@ def write_object(self, content):
         """
         raise NotImplementedError
 
-    def open(self, mode='r', encoding=None):
+    def open(self, mode='r', encoding=None, **kwargs):
         """
         Opens a OBSFile that can be read or written to and is uploaded to
         the remote service.
+
+        For examples of reading and writing opened objects, view
+        OBSFile.
+
+        Args:
+            mode (str): The mode of object IO. Currently supports reading
+                ("r" or "rb") and writing ("w", "wb")
+            encoding (str): text encoding to use. Defaults to
+                ``locale.getpreferredencoding(False)`` (Python 3 only)
+            kwargs (dict): DEPRECATED FOR SWIFT (use `stor.settings.use()`
+                instead). A dictionary of arguments that will be
+                passed as keyword args to `SwiftPath.upload` or `DXPath.upload`
+                if any writes occur on the opened resource.
+
+        Returns:
+            OBSFile: The file object for Swift/S3/DX.
+
+        Raises:
+            SwiftError: A swift client error occurred.
+            DNAnexusError: A dxpy client error occured.
+            RemoteError: A s3 client error occurred.
         """
-        raise NotImplementedError
+        swift_upload_options = kwargs.pop('swift_upload_options', {})
+        kwargs.update(**swift_upload_options)
+        return OBSFile(self, mode=mode, encoding=encoding, **kwargs)
```

Comment:
Nice!

let's just do an API break (please add this to release notes) and only support encoding argument here.

(release notes are in docs/release_notes.rst)

Are there any other breaking changes for S3 or Swift?

Reply:
What do you mean by an api break here? What do you mean by only support encoding argument? What would happen to the mode argument? 

No, I don't think other functions in S3 or Swift have been changed. There is a similar change like
{code}
swift_upload_options = kwargs.pop('swift_upload_options', {})
kwargs.update(**swift_upload_options) 
{/code}
in utils.copy/copytree though

Reply:
let's make an API break so we no longer accept `swift_upload_options` as a keyword argument and no longer accept arbitrary keyword arguments to this function (i.e., *only* the `encoding` argument)

Reply:
How can open not take a mode argument? I assume you just meant to remove `swift_upload_options` / `kwargs` here. Removed and fixed the unittest it broke